File: OOPJ/Lab/002:/BubbleSort.java
/*
   Sort an array using bubble sort.
 */

//created by akashbhalotia
import java.util.Scanner;
class Sorting
{
    private int N; //data members
    private int[] a;

    public Sorting() //constructor taking the input
    {
        Scanner sc=new Scanner(System.in);

        System.out.println("Enter array size");
        N=sc.nextInt();
        a=new int[N];

        for(int i=0;i<N;i++)
        {
            System.out.println("Enter a number");
            a[i]=sc.nextInt();
        }
    }
    public void sort() //sorts the array
    {
        int tmp;

        for(int i=0;i<N;i++)
        {
            for(int j=i;j<N-i-1;j++)
            {
                if(a[j]>a[j+1])
                {
                    tmp=a[j];
                    a[j]=a[j+1];
                    a[j+1]=tmp;
                }
            }
        }
    }
    public void display() //displays the array
    {
        for(int i=0;i<N;i++)
            System.out.print(a[i]+",");
        System.out.println();
    }
}
class Driver //driver class
{
    public static void main(String[] args)
    {
        Sorting obj=new Sorting();

        System.out.println("Array before sorting:");
        obj.display();

        obj.sort();

        System.out.println("Array after sorting:");
        obj.display();
    }
}

File: DS/LabPrograms/001: 1-D Array/BubbleSort.c
/*
 * Bubble sort. O(N^2).
 */

//created by Whiplash99
#include <stdio.h>

void input(int a[], int N) //takes input
{
    for(int i=0;i<N;i++)
    {
        printf("Enter a number\n");
        scanf("%d",&a[i]);
    }
}
void swap(int *a, int *b) //swaps the values of two numbers
{
    int tmp=*a;
    *a=*b;
    *b=tmp;
}
void sort(int a[], int N) //sorts the array
{
    for(int i=0;i<N;i++)
    {
        for(int j=0;j<N-i-1;j++)
        {
            if(a[j]>a[j+1])
                swap(&a[j],&a[j+1]);
        }
    }
}
void display(int a[], int N) //displays elements of the array
{
    for(int i=0;i<N;i++)
        printf("%d,",a[i]);
    printf("\n\n");
}
int main() //main method
{
    int N;

    printf("Enter the size of the array\n");
    scanf("%d",&N);

    int a[N];
    input(a,N);

    printf("\nArray before sorting:\n");
    display(a,N);

    sort(a,N);

    printf("Array after sorting:\n");
    display(a,N);

    return 0;
}
